File: perl/sum.pl
#!C:/xampp/perl/bin/perl.exe
#!/usr/bin/perl -w
# sum_cgi.pl

use CGI ":standard";

# get the parameter value from the brower query
my ($num1, $num2) = (param("num1"), param("num2"));

# validate and compute sume
if ($num1 =~ /^[+-]?(\d+(?:\.\d+)?)$/ &&  $num2 =~ /^[+-]?(\d+(?:\.\d+)?)$/) {
    $sum = $num1+$num2;
    $result = "<h2>$num1 + $num2 = $sum</h2>";
} else {
    $result = "<h2>Invalid input</h2>";
}

# Produce html response to browser
print header();
print start_html("Sum Caculator");
print $result;
print end_html();

File: js/sum.js
function sum() {
         var n1 = document.getElementById("num1").value;
         var n2 = document.getElementById("num2").value;
         var out = "<h2>invalid input</h2>";

         if  (!isNaN(parseFloat(n1)) && !isNaN(parseFloat(n2)))  {
            result = parseFloat(n1) + parseFloat(n2);
            out = n1 + " + " + n2 + " = " + result;
         } 
         var obj = document.getElementById("target");
         obj.innerHTML =  out;